### PR TITLE
Add marketing campaign page

### PR DESCRIPTION
**Pull request by eadpearce**
screencapture-exred-trade-great-8007-prototype-campaign-2018-11-19-18_05_01

### CODE REVIEW

File: marketing/views.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from django.views.generic import TemplateView
+
+from core.mixins import GetCMSPageMixin
+
+
+class CampaignPageView(
+    GetCMSPageMixin,
+    TemplateView
```
Comment (richtier) [inline marketing/views.py]: these could all be one line - would be easier to read as `class CampaignPageView(GetCMSPageMixin, TemplateView)`

Comment (richtier) [reply marketing/views.py]: I suspect this needs to be feature flagged

File: conf/settings.py [hunk 2]
```diff
@@ -68,6 +68,7 @@
     'export_elements',
     'directory_components',
     'prototype',
+    'marketing',
     'euexit',
```
Comment (richtier) [inline conf/settings.py]: can we be more specific - technically everything in our services is marketing (we're a wholly-owned subsidiary of the marketing department, after all)

Comment (richtier) [reply conf/settings.py]: naming is hard

Comment (richtier) [reply conf/settings.py]: as discucced - can you please move these to `core`

File: core/mixins.py [hunk 3]
```diff
@@ -16,6 +16,12 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
 
+class CampaignPagesFeatureFlagMixin(NotFoundOnDisabledFeature):
+    @property
+    def flag(self):
+        return settings.FEATURE_FLAGS['CAMPAIGN_PAGES_ON']
```
Comment (richtier) [inline core/mixins.py]: I was going to suggest that as this is not a dynamic value we can do:

```
class CampaignPagesFeatureFlagMixin(NotFoundOnDisabledFeature):
    fag = settings.FEATURE_FLAGS['CAMPAIGN_PAGES_ON']
```

**but** your way is easier to test :+1:

Comment (eadpearce) [reply core/mixins.py]: don't look at me i just copypasted one of the other feature flag mixins :'D

**Pull request merged**